src/libraries/System.Net.Sockets/src/System/Net/Sockets/UnixDomainSocketEndPoint.cs

```diff
@@ -124,21 +137,61 @@ public override string ToString()
             }
         }
 
+        [MemberNotNull(nameof(_fullPath))]
+        private void EnsurePathNormalized()
+        {
+            Debug.Assert(!IsAbstract(_path));
+
+            _fullPath ??= Path.GetFullPath(_path);
+        }
+
+        public override bool Equals([NotNullWhen(true)] object? obj)
+        {
+            if (obj is not UnixDomainSocketEndPoint uds)
+                return false;
+
+            switch ((IsAbstract(_path), IsAbstract(uds._path)))
+            {
+                case (true, true):
+                    // abstract paths are case-sensitive on Windows and Unix-based systems
+                    return MemoryExtensions.Equals(_path.AsSpan(1), uds._path.AsSpan(1), StringComparison.Ordinal);
```

Comment:
Please comment on why we skip the first character

Reply:
Is this necessary? Because it is obvious from the context. Abstract paths in Unix Domain Sockets must start from `\0` character.

Reply:
I am not versed in Unix Domain Sockets, so it wasn't obvious to me. You can consider it a mere suggestion.

Another question, did you try not skipping the first character? From looking into the `MemoryExtension.Equals` implementation, it seems to me that the method is not affected by null characters in the middle.

Reply:
I'm following the convention used across this class. My code is not the only place where slicing is used. Moreover, I'm not sure how the string comparer deals with null characters.